### PR TITLE
Fix Potential Double Delete Bug

### PR DESCRIPTION
**Pull request by ajorgensen**
This fixes a double free bug that periodically causes the stream manager to crash if other stream managers or are killed at the right time. 

In this case opkt has the potential to be freed twice in the event of a failed packet sent in both `InternalSendResponse` and `conn->sendPacket(opkt)`

### CODE REVIEW

**Comment by kramasamy** [comment]
We can merge it

**Comment by ajorgensen** [comment]
@nlu90 fixed. Is there a bazel command just run checkstyle?

**Comment by kramasamy** [comment]
@ajorgensen - here it is

bazel build --config=darwin <targets>

**Comment by ajorgensen** [comment]
@kramasamy  Oh interesting... the checkstyles never run for me during a normal build. For example if I introduce the same checkstyle issue and run

` bazel build //heron/common/src/cpp/..` 

It builds successfully. Its not until I run the `scripts/packages` directive that the checkstyle actually fires.

File: heron/common/src/cpp/network/server.cpp [hunk 1]
```diff
@@ -211,17 +207,16 @@ void Server::InternalSendRequest(Connection* _conn, google::protobuf::Message* _
   sp_int32 byte_size = _request->ByteSize();
   sp_uint32 sop = OutgoingPacket::SizeRequiredToPackString(_request->GetTypeName()) + REQID_size +
                   OutgoingPacket::SizeRequiredToPackProtocolBuffer(byte_size);
-  auto opkt = new OutgoingPacket(sop);
-  CHECK_EQ(opkt->PackString(_request->GetTypeName()), 0);
-  CHECK_EQ(opkt->PackREQID(rid), 0);
-  CHECK_EQ(opkt->PackProtocolBuffer(*_request, byte_size), 0);
+  OutgoingPacket opkt(sop);
```
Comment (nwangtw) [inline heron/common/src/cpp/network/server.cpp]: Suggest to make the same change in SendResponse() and send SendMessage() in this file.

**Comment by nicknezis** [comment]
@nwangtw It's been a while, but is this still something that should be merged in?

**Pull request closed without merging**